prow/istio-presubmit.sh

```diff
@@ -64,6 +64,29 @@ else
 fi
 cd $ROOT
 
+# go test setup
+go get -u github.com/golang/dep/cmd/dep
+time dep ensure -v
+#mkdir -p ~/.kube
+#cp ./.circleci/config ~/.kube/config
+#ln -sf ~/.kube/config ./pilot/platform/kube/config
+mkdir -p ~/envoy
+cd ~/envoy
+ISTIO_PROXY_BUCKET=$(sed 's/ = /=/' <<< $( awk '/ISTIO_PROXY_BUCKET =/' $ROOT/WORKSPACE))
+PROXYVERSION=$(sed 's/[^"]*"\([^"]*\)".*/\1/' <<<  $ISTIO_PROXY_BUCKET)
+PROXY=debug-$PROXYVERSION
+wget -qO- https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz | tar xvz
+ln -sf ~/envoy/usr/local/bin/envoy $ROOT/pilot/proxy/envoy/envoy
+
+cd $ROOT
+
+# go test execution
+echo FIXME remove mixer tools exclusion after tests can be run without bazel
```

Comment:
Curious, why do need to run both `go test` and `bazel test`?

Reply:
My understanding is, after talking to @mandarjog, is that we want to support Go and bazel in case proto files are edited. Just running `bazel test` is not good enough because `go test` works on checked in files whereas `bazel test` does not. Therefore, there can be a case where `bazel test` works and not `go test` (which is the current state and circle Ci is broken because someone forgot to update the .pb.go checked in files)

Reply:
Ah, I see. The bigger problem seems to be mixed use of committed and generated *.pb.go files.